src/components/common/Interactive.tsx

```diff
@@ -57,12 +71,15 @@ const InteractiveBase = ({ onMove, onKey, ...rest }: Props) => {
       preventDefaultMove(nativeEvent);
 
       if (isInvalid(nativeEvent, hasTouch.current) || !el) return;
-      hasTouch.current = isTouch(nativeEvent);
+      if (isTouch(nativeEvent)) {
+        hasTouch.current = true;
+        touchId.current = nativeEvent.changedTouches[0].identifier;
```

Comment:
Didn't know about `changedTouches` property 🤓
Hope it works everywhere we need.